flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -147,7 +147,7 @@ public MetricRegistryImpl(MetricRegistryConfiguration config, Collection<Reporte
 					}
 					reporters.add(reporterInstance);
 
-					String delimiterForReporter = reporterSetup.getDelimiter().orElse(String.valueOf(globalDelimiter));
```

Comment:
It seems we continue to use the `globalDelimiter` in `AbstractMetricGroup` for some purposes.
Is it supposed to be so or is it another story with logical scopes?

Reply:
the abstract metric group can be used a) by reporters, in which case we rely on the reporter defined here, and b) user-functions, in which case they still rely on the global delimiter.